### PR TITLE
tidy build config after Gradle & Kotlin updates

### PR DESCRIPTION
**Pull request by aSemy**
- enable Gradle config cache (Kotlin 1.9 is now compatible)
- update Gradle config cache property names
- use Kotlin default hierarchy https://kotlinlang.org/docs/multiplatform-hierarchy.html#default-hierarchy-template
- fix Kotlin config deprecations

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -23,7 +23,7 @@ kotlin {
             kotlin.srcDirs(tasks.yamlTestSuiteDataSources)
 
             dependencies {
-                implementation(platform("io.kotest:kotest-bom:5.8.0"))
+                implementation(project.dependencies.platform("io.kotest:kotest-bom:5.8.0"))
```
Comment (krzema12) [inline build.gradle.kts:26]: Why this change? The former was more compact.

File: buildSrc/src/main/kotlin/buildsrc/conventions/lang/kotlin-multiplatform-native.gradle.kts [hunk 2]
```diff
@@ -51,52 +24,4 @@ kotlin {
     //iosSimulatorArm64()
     //tvosSimulatorArm64()
     //watchosSimulatorArm64()
-
-    @Suppress("UNUSED_VARIABLE")
```
Comment (krzema12) [inline buildSrc/src/main/kotlin/buildsrc/conventions/lang/kotlin-multiplatform-native.gradle.kts:55]: :tada:

File: build.gradle.kts (hunk 1 above)
Comment (aSemy) [reply build.gradle.kts:26]: It was deprecated

https://github.com/JetBrains/kotlin/blob/v1.9.21/libraries/tools/kotlin-gradle-plugin-api/src/common/kotlin/org/jetbrains/kotlin/gradle/plugin/HasKotlinDependencies.kt#L76-L81

**Review by krzema12 — APPROVED** [review]
